File: gaia-sdk-javascript/src/graphql/request/type/Role.ts


import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Represents Role information
 */
export class Role extends Array<(_:VariableRegistry) => string> {
public _typeName = "Role";
    /**
     * Id of the tenant
     */
    public tenantId = () => { 
        this.push(_ => "tenantId")
    };

    /**
     * Id of the role
     */
    public roleId = () => { 
        this.push(_ => "roleId")
    };

    /**
     * The name of the role
     */
    public name = () => { 
        this.push(_ => "name")
    };

    /**
     * The permissions of the role
     */
    public permissions = () => { 
        this.push(_ => "permissions")
    };

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

File: gaia-sdk-python/gaia_sdk/graphql/request/type/Role.py


from typing import Callable, List
from gaia_sdk.api.VariableRegistry import VariableRegistry
from gaia_sdk.graphql.request.enumeration.Order import Order
from gaia_sdk.graphql.request.enumeration.OrderByField import OrderByField
from gaia_sdk.graphql.request.enumeration.EdgeOrderByField import EdgeOrderByField
from gaia_sdk.graphql.request.enumeration.EdgeType import EdgeType


class Role(list):
    """
    Represents Role information
    """

    """
    Id of the tenant
    """
    def tenant_id(self):
        self.append(lambda x: "tenantId")

    """
    Id of the role
    """
    def role_id(self):
        self.append(lambda x: "roleId")

    """
    The name of the role
    """
    def name(self):
        self.append(lambda x: "name")

    """
    The permissions of the role
    """
    def permissions(self):
        self.append(lambda x: "permissions")

    def render(self, registry: VariableRegistry):
        return " ".join(map(lambda e: e(registry), self))
